lib/api/bucketGetTagging.js

```diff
@@ -0,0 +1,105 @@
+const { metadataValidateBucket } = require('../metadata/metadataUtils');
+const util = require('node:util');
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+const { checkExpectedBucketOwner } = require('./apiUtils/authorization/bucketOwner');
+const { pushMetric } = require('../utapi/utilities');
+const { errors, s3middleware } = require('arsenal');
+const escapeForXml = s3middleware.escapeForXml;
+
+//  Sample XML response:
+/*
+    <Tagging>
+        <TagSet>
+            <Tag>
+                <Key>string</Key>
+                <Value>string</Value>
+            </Tag>
+            <Tag>
+                <Key>string</Key>
+                <Value>string</Value>
+            </Tag>
+        </TagSet>
+    </Tagging>
+*/
+
+/**
+ * Convert Versioning Configuration object of a bucket into xml format.
+ * @param {array} tags - set of bucket tag keys and values
+ * @return {string} - the converted xml string of the versioning configuration
+ */
+function convertToXml(tags) {
+    const xml = [];
+
+    xml.push('<?xml version="1.0" encoding="UTF-8" standalone="yes"?>',
+        '<Tagging>',
+        '<TagSet>'
+    );
+
+    tags.forEach(tag => {
+        xml.push('<Tag>');
+        xml.push(`<Key>${escapeForXml(tag.Key)}</Key>`);
+        xml.push(`<Value>${escapeForXml(tag.Value)}</Value>`);
+        xml.push('</Tag>');
+    });
+
+    xml.push('</TagSet>',
+        '</Tagging>');
+
+    return xml.join('');
+}
+
+/**
+ * bucketGetVersioning - Return Versioning Configuration for bucket
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to respond to http request
+ *  with either error code or xml response body
+ * @return {undefined}
+ */
+async function bucketGetTagging(authInfo, request, log, callback) {
+    log.debug('processing request', { method: 'bucketGetTagging' });
+
+    const { bucketName, headers } = request;
+
+    const metadataValParams = {
+        authInfo,
+        bucketName,
+        requestType: 'bucketGetTagging',
+        request,
+    };
+
+    let bucket;
+    let xml = null;
+
+    try {
+        let promisifyFunc = util.promisify(metadataValidateBucket);
```

Comment:
I would create a separate value for each promisified(?) function, maybe even at the top level if you want.
Such as `const metadataValidateBucketPromise = util.promisify(...);`

Reply:
done https://github.com/scality/cloudserver/pull/4481/commits/96157ec37c9e4398618eb993ff7e5fb92c4e8141#diff-889d6969db5c5fda428f1b3f508e237cb4bc1419949c78bc0a146bcca7d9209bR66